keps/sig-api-machinery/5000-api-linting-crd-schema-tooling/README.md

```diff
@@ -0,0 +1,522 @@
+| anyOf | Property | Validates compatibility of changes to the anyOf constraint of a property. Net new and removal of anyOf constraints are flagged | No | Yes |
+| not | Property | Validates compatibility of changes to the not constraint of a property. Net new not constraints are flagged | No | Yes |
+
+Proposed validations are tracked via https://github.com/everettraven/crd-diff/issues/3.
+All validations have some form of configuration option. For more information on the configuration options that exist for each validation, see https://everettraven.github.io/crd-diff/#/validations
+This existing configuration approach is not ideal. It is highly repetitive and allows for configuration options that may not actually make sense. We should be more opinionated
+than this existing approach is. A new approach for configuration is proposed further down in this section.
+
+Validations in https://github.com/openshift/crd-schema-checker  both existing and proposed:
+
+| Name | Scope | Description | Implemented | Carry |
+| ---- | ----- | ----------- | ----------- | ----- |
+| NoBools | Property | Validates that CRD properties are not of type `boolean` | Yes | No, already implemented as a linter in `kal` |
+| NoFloats | Property | Validates that CRD properties are not of type `number` | Yes | No, already implemented as a linter in `kal` |
+| NoUints | Property | Validates that CRD properties are not of type `uint` | Yes | No, already implemented as a linter in `kal` |
+| NoFieldRemoval | CRD | Validates that existing fields are not removed from a CRD | Yes | Yes |
+| NoEnumRemoval | Property | Validates that existing enum values are not removed from a property | Yes | Yes |
+| NoMaps | Property | Validates that CRD properties of type `object` do not have an `additionalProperties` field specified | Yes | No, if desired this is better suited for a linter in `kal` |
+| NoDataTypeChange | Property | Validates that property type is not changed | Yes | Yes |
+| MustHaveStatus | CRD | Validates that the CRD has a status subresource | Yes | No, this is already implemented as a linter in `kal` |
+| ListsMustHaveSSATags | Property | Validates that lists have the `x-kubernetes-list-type` tag for server side apply | Yes | No, this is desired as a linter in `kal` |
+| ConditionsMustHaveSSATags | Property | Validates that status conditions fields have the appropriate tags for server side apply | Yes | No, this is already implemented as a subset of the `conditions` linter in `kal` |
+| NoNewRequiredFields | CRD | Validates that no new required fields are added | Yes | Yes |
+| MustNotExceedCostBudget | Property | Validates that `XValidations` don't exceed the CEL cost budget | Yes | No, if desired this is better suited for a linter in `kal` |
+
+#### Validation configuration
+
+The proposed approach for configuration of individual validations when using as a CLI is to use a configuration file.
```

Comment:
Doesn't golangci-lint already have extensible plugin config in a yaml file?

https://golangci-lint.run/plugins/go-plugins/#configure-a-plugin
https://golangci-lint.run/usage/linters/#custom